scripts/requirements: Split out doc requirements into their own file

galak opened the pull request:
Split the requirements.txt so we can manage different uses.  For
example if you don't need to build the docs you can ignore the
requirements in requirements-doc.txt, or if you ONLY need to build
the docs (for example CI doc job) you can just pay attention to
what's in requirements-doc.txt.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/requirements: Split out doc requirements into their own file

Split the requirements.txt so we can manage different uses.  For
example if you don't need to build the docs you can ignore the
requirements in requirements-doc.txt, or if you ONLY need to build
the docs (for example CI doc job) you can just pay attention to
what's in requirements-doc.txt.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mbolivar reviewed (changes requested):
Please update doc/README.rst

galak commented:
> Please update doc/README.rst

done.

mbolivar reviewed (dismissed):
I think that's not the right update

mbolivar commented on doc/README.rst:
```diff
@@ -117,7 +117,7 @@ Sphinx supports easy customization of the generated documentation
 another ``make htmldocs`` and the output layout and style is changed.
 The ``read-the-docs`` theme is installed as part of the
-``requirements.txt`` list above, and will be used if it's available, for
+``requirements-doc.txt`` list above, and will be used if it's available, for
```
There's no mention of using pip3 install -r requirements-doc.txt above; there should be if this PR is going to go in

galak commented:
> I think that's not the right update

What update are you wanting me to make?

mbolivar commented:
> > I think that's not the right update
> 
> What update are you wanting me to make?

I left more specifics above. Please read through the file and make sure the instructions still work.

galak commented:
> I left more specifics above. Please read through the file and make sure the instructions still work.

The requirements.txt is changed in such a way that its not visible.  The question is do we want to convey to users to do something different.  I'm wanting to break up requirements.txt mostly for CI purposes, but also to help a bit with documenting what python packages are for what purpose.

If a user still does `pip3 install --user -r ~/zephyrproject/zephyr/scripts/requirements.txt` they still get the exact some set of packages installed before or after this PR.

I'm happy to make changes to the docs, just want to make sure they are changes we want to make.

mbolivar commented:
> I'm happy to make changes to the docs, just want to make sure they are changes we want to make.

Ah, OK, I'm complaining about a bug that wasn't introduced by this PR, then. Looks like the doc README is out of date and the requirements.txt step mentioned there is stale.

Maybe tack this on if you feel like it?

```
diff --git a/doc/README.rst b/doc/README.rst
index ab3073a4b3..cc29fbf870 100644
--- a/doc/README.rst
+++ b/doc/README.rst
@@ -114,12 +114,11 @@ Documentation presentation theme
 ********************************
 
 Sphinx supports easy customization of the generated documentation
-appearance through the use of themes.  Replace the theme files and do
-another ``make htmldocs`` and the output layout and style is changed.
+appearance through the use of themes. Replace the theme files and do another
+``make htmldocs`` and the output layout and style is changed.
 The ``read-the-docs`` theme is installed as part of the
-``requirements-doc.txt`` list above, and will be used if it's available, for
-local doc generation.
-
+:ref:`install_py_requirements` step you took in the getting started
+guide.
 
 Running the documentation processors
 ************************************
diff --git a/doc/getting_started/index.rst b/doc/getting_started/index.rst
index caff3edbfa..d56b938c35 100644
--- a/doc/getting_started/index.rst
+++ b/doc/getting_started/index.rst
@@ -246,6 +246,8 @@ directory using west:
          cd zephyrproject
          west update
 
+.. _install_py_requirements:
+
 .. rst-class:: numbered-step
 
 Install needed Python packages
```

I can open a separate PR if you want. Dropping the -1 for now.

galak committed:
doc: Add reference in toplevel README.rst to install_py_requirements

In the section 'Documentation presentation theme' put a reference to
'install_py_requirements' section in the getting start guide.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
WIP: sort requirements into things that use it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@mbolivar @nashif would appreciate your help in identifying why we need some of the python packages (like `wheel`, `git-spindle>=2.0`, and `Pillow`.

I also want to split things up to try and have some requirements files that CI can utilize to install the min set of python packages.

Maybe one breakdown would be:
1. what you need to compile/create images (west, pyelftools, intelhex, packaging, windows-curses)
2. what you need to create docs (breathe, docutils, sphinx*)
3. what you need to build test (sanitycheck, pytest, gcovr, ply, pykwalify,  etc).
4. what you need to run test (pyserial, pyocd, tabulate, anytree??)
5. extras (junit2html, gitlint)

Than CI can utilize 1 - 3 as needed.

galak commented:
@henrikbrixandersen commented that Pillow is used by `scripts/gen_cfb_font_header.py`

galak commented:
Closing in favor of #22635 and #22633
